internals/types/index.d.ts

```diff
@@ -510,7 +510,33 @@ export interface Token {
   teamId?: string;
 }
 
-export interface GitMetadata {
+export type CiMetadata =
+  | {
+      ci: true;
+      ciType: 'github-actions';
+      ciGitProviderUsername: string;
+      ciGitRepoVisibility?: never;
+    }
+  | {
+      ci: true;
+      ciType: 'gitlab-ci-cd';
+      ciGitProviderUsername: string;
+      ciGitRepoVisibility?: 'public' | 'private' | 'internal';
+    }
+  | {
+      ci: true;
+      ciType: 'bitbucket-pipelines';
+      ciGitProviderUsername?: never;
+      ciGitRepoVisibility?: never;
+    }
+  | {
+      ci?: true;
+      ciType?: never;
```

Comment:
should this be unknown. feels like empty would either break or make the discriminated union hard to use. not positive on that one.